lib/routes/utils/request_helper.js

```diff
@@ -80,8 +80,56 @@ function shouldSendVerifyAccountEmail(account, request) {
   return (!context || !!sendEmailIfUnverified) && !account.emailVerified
 }
 
+/**
+ * Returns whether or not to use signin unblock feature on a request.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninUnblock(account, config, request) {
+
+  var enabled = config.signinUnblock && config.signinUnblock.enabled
+  if (!enabled) {
+    return false
+  }
+
+  //if (request.app.isSuspiciousRequest) {
+  //  return true
+  //}
+
+  // Or if the email address matching one of these regexes.
+  var email = account.email
+  var isValidEmail = config.signinUnblock.forceEmailRegex.some(function (reg) {
+    var emailReg = new RegExp(reg)
+    return emailReg.test(email)
+  })
+
+  if (isValidEmail) {
+    return true
+  }
+
+  // While we're testing this feature, there may be some funky
+  // edge-cases in device login flows that haven't been fully tested.
+  // Temporarily avoid them for regular users by checking the `context` flag,
+  // This check may go away in the final version of this feature.
+  var context = request.payload && request.payload.metricsContext && request.payload.metricsContext.context
+  var isValidContext = context && (config.signinUnblock.supportedClients.indexOf(context) > -1)
+  if (!isValidContext) {
+    return false
+  }
+
+  // Check to see if user in roll-out cohort.
+  // Cohort is determined by user's uid.
+  var uid = account.uid.toString('hex')
+  var uidNum = parseInt(uid.substr(0, 4), 16) % 100
+  return uidNum < (config.signinUnblock.sample_rate * 100)
```

Comment:
This will have to change for the reasons highlighted by @rfk in https://github.com/mozilla/fxa-auth-server/pull/1470#discussion_r80139482. It can do so by making a call to `isSampledUser` if this gets merged after that PR, or I can refactor mine to do that if your PR gets merged first.